agent/api/task.go

```diff
@@ -190,6 +192,13 @@ func (task *Task) DockerConfig(container *Container) (*docker.Config, *DockerCli
 }
 
 func (task *Task) dockerConfig(container *Container) (*docker.Config, *DockerClientConfigError) {
+	image := container.Image
+	if strings.HasPrefix(container.Image, "s3://") {
+		hash := md5.New()
+		tag := hex.EncodeToString(hash.Sum([]byte(container.Image)))
```

Comment:
We would have to do that every time the container changes state (starts, stops, etc), or add another place to store it within the container.